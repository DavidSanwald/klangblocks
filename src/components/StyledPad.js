import styled, { css } from 'styled-components'
import React from 'react'
import sample from 'lodash.sample'
import { darken } from 'polished'
import PropTypes from 'prop-types'
import { background } from '../config/styles'


const StyledPad = (props) => (
  <StyledSquare {...props}
      width="10"
      height="10"
      x={props.n*10}
      y={props.m*10}
      state={props.state}

      onClick={props.onClick}
    />
);

const StyledSquare = styled.rect`
      cursor: pointer;
${props => {
  switch (props.state) {
    case 'playing':
      return css`
    fill: ${sample(background.playing)};
    transition: fill ${props.timing} cubic-bezier(0.175, 0.885, 0.32, 1.275);
  `
    case 'idle':
      return css`
    fill: ${background.idle};
    transition: fill ${props.timing} cubic-bezier(0.175, 0.885, 0.32, 1.275);

`
    case 'selected':
      return css`
     fill: ${darken(0.02, background.idle)};
     transition: fill ${props.timing} cubic-bezier(0.175, 0.885, 0.32, 1.275);
     `
    default:
      return css`
      fill: ${background.idle};
     `
    }
  }};
  width:  ${props => props.width};
  max-width: ${props => props.maxwidth};
  height: ${props => props.height};
  cursor: pointer;
  &:hover {
    background: ${darken(0.01, background.idle)};
    }
    `
  StyledSquare.propTypes = {
    state: PropTypes.oneOf(['idle', 'playing', 'selected']).isRequired,
    onClick: PropTypes.func.isRequired,
    width: PropTypes.string,
    maxwidth: PropTypes.string,
    height: PropTypes.string,
    timing: PropTypes.string,
  }


  StyledSquare.defaultProps = {
    width: '8vh',
    maxwidth: '12.5%',
    height: '8vh',
    timing: '0.5s' ,
  }



export default StyledPad
