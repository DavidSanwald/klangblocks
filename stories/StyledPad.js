import StyledPad from './components/StyledPad'
