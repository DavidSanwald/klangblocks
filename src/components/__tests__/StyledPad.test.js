import React from 'react'
import renderer from 'react-test-renderer'
import { matcher, serializer } from 'jest-styled-components'
import StyledPad from '../StyledPad'

expect.extend(matcher)
expect.addSnapshotSerializer(serializer)

// foo.js
module.exports = function() {
  // some implementation;
};

// test.js
jest.mock('lodash.sample'); // this happens automatically with automocking
const sample = require('lodash.sample');

// foo is a mock function
sample.mockImplementation((array) => array[0]);
const spyOnClick = jest.fn();


it('renders without crashing', () => {
  const tree = renderer.create(<StyledPad
    state='playing' onClick={spyOnClick}  />).toJSON()
  expect(tree).toMatchStyledComponentsSnapshot()
})
