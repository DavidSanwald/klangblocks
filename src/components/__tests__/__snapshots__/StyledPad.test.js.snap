// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.rlkeg {
  cursor: pointer;
  fill: #67001f;
  transition: fill 0.5s cubic-bezier(0.175, 0.885, 0.32, 1.275);
  -webkit-transition: fill 0.5s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}

.rlkeg:hover {
  opacity: 0.9;
}

<rect
  className="rlkeg"
  height="10"
  onClick={[Function]}
  width="10"
  x={NaN}
  y={NaN}
/>
`;
